### DISCUSSION TITLE
Plus in filename gets displayed as space character and then cannot delete the file

### DISCUSSION THREAD

**Discussion by jamshid** [iterate-ch/cyberduck#15265] [Q&A] [answered]
**Describe the bug**
With at least macOS Cyberduck Version 8.7.0 (40629) uploading a file with a name that includes a space e.g. `cat+mustache.gif` the file will upload correctly but displays as `cat mustache.gif`. Then deleting the file fails to do anything. 

**To Reproduce**
1. Open an S3 connection and go into a bucket.
2. Create a file named `cat+mustache.gif` on your file system.
3. Drag it into the Cyberduck bucket window, it will upload.
4. See that it is incorrectly displayed as `cat mustache.gif`.

**Expected behavior**
A plus character `+` should be encoded as `%2B` in a url. A `+` character should only mean space ` ` when used in a query arg.

**Screenshots**
<img width="805" alt="Screenshot 2023-10-24 at 1 35 13 PM" src="https://github.com/iterate-ch/cyberduck/assets/8549/817dc3c2-7258-4c81-ba18-c35d97e77899">

FWIW this is what is returned from bucket listing api of the S3-compatible system:
```
<?xml version="1.0" encoding="UTF-8"?><ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Name>jamtest</Name><Prefix/><MaxKeys>1000</MaxKeys><Marker/><IsTruncated>false</IsTruncated><Contents><Key>cat+mustache.gif</Key><LastModified>2023-10-24T18:03:12.216Z</LastModified><ETag>"549556bdbba8ce1dd01d4980c4e816a6"</ETag><Size>919069</Size><Owner><ID>dcadmin</ID><DisplayName>dcadmin</DisplayName></Owner><StorageClass>STANDARD</StorageClass></Contents></ListBucketResult>
```
**Desktop (please complete the following information):**
 - OS: macOS Sonoma 
 - Version 14.0

**Log Files**
[cyberduck.log](https://github.com/iterate-ch/cyberduck/files/13125657/cyberduck.log)

**Additional context**
I'd be surprised if this is a general bug not tested or seen before with S3. Maybe AWS encodes listings slightly different?

**Comment by dkocher** [comment, marked as answer]
This cannot be reproduced when connected to AWS S3. You will find `encoding-type=url` in the GET request to list objects [^1].

> Requests Amazon S3 to encode the object keys in the response and specifies the encoding method to use. An object key can contain any Unicode character; however, the XML 1.0 parser cannot parse some characters, such as characters with an ASCII value from 0 to 10. For characters that are not supported in XML 1.0, you can add this parameter to request that Amazon S3 encode the keys in the response. Valid Values: `url`

Sample request and response

```
PUT /cat%2Bmustache.gif HTTP/1.1
Content-Type: image/gif
x-amz-content-sha256: 322d70b741131243b52e1c09ac9cb1a8308c20d5e4f4c217bcae2aa0f5493343
x-amz-storage-class: STANDARD
Content-Length: 6
Host: test-cyberduck-15264.s3.dualstack.eu-west-3.amazonaws.com
Connection: Keep-Alive
User-Agent: Cyberduck/8.7.0.40629 (Mac OS X/14.0) (aarch64)
Accept-Encoding: gzip,deflate
x-amz-date: 20231024T194005Z

HTTP/1.1 200 OK
x-amz-id-2: K7yLWyZhkx5ttFVR8KjZT9oEroCZErZ29jcHzckVcx0axFp7NAydWKT/zJzsyBQ3IMD0myumsdQ=
x-amz-request-id: P6VCRDGJZ3REZVAH
Date: Tue, 24 Oct 2023 19:40:09 GMT
x-amz-server-side-encryption: AES256
ETag: "71408e8b54e267188ed931e54e65a1a9"
Server: AmazonS3
Content-Length: 0

GET /?encoding-type=url&max-keys=1000&prefix=&delimiter=%2F HTTP/1.1
Host: test-cyberduck-15264.s3.dualstack.eu-west-3.amazonaws.com
Connection: Keep-Alive
User-Agent: Cyberduck/8.7.0.40629 (Mac OS X/14.0) (aarch64)
Accept-Encoding: gzip,deflate
x-amz-request-payer: requester
x-amz-date: 20231024T194008Z
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855

HTTP/1.1 200 OK
x-amz-id-2: 6lPvc3Hy0rB7RyUouBIH0wknMcIc8DhS2gitMCQBOrzOmHawmlLfYmPrzdr/AfSrcSM7Z4eSQss=
x-amz-request-id: P6V8S2YXRG5AMPB7
Date: Tue, 24 Oct 2023 19:40:09 GMT
x-amz-bucket-region: eu-west-3
Content-Type: application/xml
Transfer-Encoding: chunked
Server: AmazonS3

<?xml version="1.0" encoding="UTF-8"?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
    <Name>test-cyberduck-15264</Name>
    <Prefix/>
    <Marker/>
    <MaxKeys>1000</MaxKeys>
    <Delimiter>/</Delimiter>
    <EncodingType>url</EncodingType>
    <IsTruncated>false</IsTruncated>
    <Contents>
        <Key>cat%2Bmustache.gif</Key>
        <LastModified>2023-10-24T19:40:09.000Z</LastModified>
        <ETag>"71408e8b54e267188ed931e54e65a1a9"</ETag>
        <Size>6</Size>
        <Owner>
            <ID>80b9982b7b08045ee86680cc47f43c84bf439494a89ece22b5330f8a49477cf6</ID>
        </Owner>
        <StorageClass>STANDARD</StorageClass>
    </Contents>
</ListBucketResult>
```


[^1]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html#API_ListObjects_RequestSyntax

**Reply by jamshid** [reply]
Thanks ok that makes sense will file a bug with the S3-compatible system to support `?encoding-type=url`.

**Reply by jamshid** [reply]
Btw it would be nice if Cyberduck used the presence of `<EncodingType>url</EncodingType>` in the listing response body to determine whether the S3-compatible system implements `?encodoing-type=url`. Other clients and SDKs seem to do this. But I did file a bug for this with the storage vendor and it will be fixed in an upcoming release. Thanks again for identifying the problem.
